### ISSUE TITLE
containerd driver doesn't support passing args without command

### ISSUE THREAD

**Issue by Oloremo** [Roblox/nomad-driver-containerd#76]
In the **docker** driver, **args** are optional but they could be passed as with **command** and without **command**:
https://www.nomadproject.io/docs/drivers/docker#args

Without command, they will be passed to entrypoint script I guess.

**contained** driver produces an error if args are passed without command:

> rpc error: code = Unknown desc = Error in creating container: Command is empty. Cannot set --args without --command.

This is an incompatibility with the docker driver and could block the migration.

Do you think it makes sense to make args behave in the same way as in docker driver?

**Comment by shishir-a412ed** [comment]
@Oloremo I am still looking into this.

[`OCI Runtime Spec`](https://github.com/opencontainers/runtime-spec/blob/master/schema/config-schema.json) doesn't have a notion of `entrypoint` (`entrypoint` is docker specific and `containerd-driver` works with OCI images), it only works with `command` and `args`, and that's why the behavior.

Need some more brainstorming from my side. Will provide an update soon.

**Comment by lisongmin** [comment]
It seems specified `args` without `command` is valid from this example:

https://github.com/opencontainers/runtime-spec/blob/master/config.md#example-linux-1

So i think we can remove this limitation.

**Comment by shishir-a412ed** [comment]
@lisongmin Thank you for the pointer, that works spot on!

1. I have removed the check. Now the user can pass `args` without setting `command` in the `containerd-driver`
2. I have also added support for `entrypoint` which can be used to override the default ENTRYPOINT defined in the image. One thing I want to call out is `entrypoint` in [`Nomad docker driver`](https://www.nomadproject.io/docs/drivers/docker#entrypoint) is a `[]string` while for `containerd-driver` is a `string`. The reason for that is, nomad docker driver interacts directly with docker HTTP API, where `entrypoint` is defined as a `[]string`. However I wanted to keep the behavior similar to docker CLI where `--entrypoint` is a `string` flag.

I am working on the tests for the PR. Once reviewed and merged, this should be available to use.

/cc @Oloremo
